py-polars/polars/internals/dataframe/frame.py

```diff
@@ -5543,10 +5545,9 @@ def lazy(self: DF) -> pli.LazyFrame:
     def select(
         self: DF,
         exprs: (
-            str
-            | pli.Expr
-            | pli.Series
-            | Iterable[str | pli.Expr | pli.Series | pli.WhenThen | pli.WhenThenThen]
+            PolarsExprType
+            | PythonLiteral
+            | Iterable[PolarsExprType | PythonLiteral]
             | None
         ) = None,
```

Comment:
I think it makes sense to add `str` here separately to indicate that it has a unique behaviour. Even though  `str` is already part of `PythonLiteral` and it's not necessary for type checking purposes, it can be nice for documentation purposes.